lib/ecto/adapters/sql.ex

```diff
@@ -355,6 +355,42 @@ defmodule Ecto.Adapters.SQL do
     opts
   end
 
+  defmacrop safe_fragment(string) do
+    quote do
+      fragment(unquote(string))
+    end
+  end
+
+  @doc """
+  Check if the given `table` exists.
+
+  Returns `true` if the `table` exists in the `repo`, otherwise `false`.
+  """
+  @spec table_exists?(Ecto.Repo.t, table :: String.t) :: boolean
+  def table_exists?(repo, table) when is_atom(repo) do
+    table_exists?(repo, Ecto.Adapter.lookup_meta(repo), table)
+  end
+
+  defp table_exists?(repo, %{sql: sql} = adapter_meta, table) do\
+    import Ecto.Query, only: [from: 2]
+
+    query = from(
+      t in "tables",
+      prefix: "information_schema",
+      where: t.table_name == ^table,
+      select: t.table_name
+    )
+    {sql_query, params} = to_sql(:all, repo, query)
+    sql_query = "#{sql_query} AND table_schema = #{sql.current_prefix_function()} LIMIT 1;"
```

Comment:
@josevalim I've finally made it work, but it's ugly 🤣 

I couldn't make `sql.current_prefix_function()` works with `fragment`. The `fragment` doesn't like dynamic strings to prevent SQL injections, but I couldn't find any option to tell it that's fine. It's a trusted function.

Do you know how we can improve this part? Any ideas?

Reply:
You can ask the adapter to return the whole query, this way you don't need to compose it. :)

Reply:
Sure! Thank you. Refactored. ✨